include/envoy/http/async_client.h

```diff
@@ -205,6 +205,10 @@ class AsyncClient {
       parent_context = v;
       return *this;
     }
+    StreamOptions& setMetadata(const envoy::config::core::v3::Metadata& m) {
```

Comment:
I think this should be `setMetadataMatch` and the corresponding field `metadata_match` to avoid confusion.

Reply:
Actually, on second thought, it's correct because it's just setting the dynamic metadata and making use of the fact that we can override metadata match criteria that way. Maybe there should just be comment somewhere noting that subset LB metadata can be passed in via this method via the envoy.lb namespace.

Reply:
👌 I will add a comment to the `setMetadata` method.